docker-entrypoint.sh

```diff
@@ -190,61 +198,13 @@ EOPHP
     chown www-data:www-data -R /var/lime/sessions
     chmod ug=rwx -R /var/lime/sessions
 
-    DBSTATUS=$(TERM=dumb php -- "$LIMESURVEY_DB_HOST" "$LIMESURVEY_DB_USER" "$LIMESURVEY_DB_PASSWORD" "$LIMESURVEY_DB_NAME" "$LIMESURVEY_TABLE_PREFIX" "$MYSQL_SSL_CA" <<'EOPHP'
-<?php
-// database might not exist, so let's try creating it (just to be safe)
-
-error_reporting(E_ERROR | E_PARSE);
-mysqli_report(MYSQLI_REPORT_OFF);
 
-$stderr = fopen('php://stderr', 'w');
-$host = $argv[1];
-$socket = null;
-if (str_contains($host, ':')) {
-    list($host, $socket) = explode(':', $argv[1], 2);
-}
-$port = 0;
-if (is_numeric($socket)) {
-    $port = (int) $socket;
-    $socket = null;
-}
-
-do {
-    $con = mysqli_init();
-    if (isset($argv[6]) && !empty($argv[6])) {
-	    mysqli_ssl_set($con,NULL,NULL,"/var/www/html/" . $argv[6],NULL,NULL);
-    }
-    $mysql = mysqli_real_connect($con,$host, $argv[2], $argv[3], '', $port, $socket, MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT);
-        if (!$mysql) {
-                fwrite($stderr, "MySQL Connection Error will retry in 3 seconds...\n");
-                sleep(3);
-        }
-} while (!$mysql);
-
-if (!$con->query('CREATE DATABASE IF NOT EXISTS `' . $con->real_escape_string($argv[4]) . '`')) {
-        fwrite($stderr, "\n" . 'MySQL "CREATE DATABASE" Error: ' . $con->error . "\n");
-        $con->close();
-        exit(1);
-}
-
-$stat = $con->select_db($con->real_escape_string($argv[4]));
-
-$inst = $con->query("SHOW TABLES LIKE '" . $con->real_escape_string($argv[5]) . "users'");
-
-$nrows = $inst->num_rows;
-
-$con->close();
-
-if ($nrows > 0) {
-        exit("DBEXISTS");
-} else {
-        exit(0);
-}
-
-EOPHP
-)
+    # The following 4 lines are borrowed from martialblog's entrypoint.sh
+    # Check if LimeSurvey database is provisioned
+    echo 'Info: Check if database already provisioned. Nevermind the stack trace.'
+    php application/commands/console.php updatedb || MUST_CREATE_DB=true
```

Comment:
this change here is neat, but it introduces a functional change. It will just attempt to create the database and then fail/exit if there is no db connection. Previously it would loop over and then wait 5 seconds and try again. This makes sure the container doesn't fail while waiting for a DB connection.

Reply:
Point well taken. In martialblog's entrypoint script, this is taken care of in a loop of netcat calls (`until nc -z -v -w30 "$DB_HOST" "$DB_PORT"...`), which is agnostic of the actual database used. This is still different from your original approach which tests if there actually is a limesurvey database table. I'll be happy to change the script to deal with PG (however, no expert at all in PHP connections to PG, but could come up with code to review). Just wanted to through this in as it reduces complexity in my opinion. Your advice/opinion is more than welcome :smile:

Reply:
thanks @bovender . I'm happy to use martialblogs approach for this - can you please update on your end and I'll test it here?